---
title: "explore male bees"
author: "Nurit Eliash"
date: "4/14/2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### load libraries
```{r}
library(tidyverse)
library(dplyr)
library(ggplot2)
library("vcfR") # for extracting genotype data from a vcf file
library(pinfsc50)
library(reshape)
library(reshape2)
library(hrbrthemes)
library(viridis)
library("plotly") # for the 3d plots
```

### Heterozygosity and inbreeding coefficient per individual
```{r}
ind_het <- read_delim("/Users/nuriteliash/Documents/GitHub/Variant_Calling_bees/data/vcf_stats/bee.het", delim = "\t",
           col_names = c("ind","homo_ob", "homo_ex", "nsites", "f"), skip = 1)

# add the sex of each sample, and the proportion of homosyzgotic and hetero sites
sex <- read.csv("/Users/nuriteliash/Documents/GitHub/Variant_Calling_bees/data/meta.csv")
ind_het <- left_join(ind_het,sex, by ="ind") %>%
  mutate(hom_prop =  homo_ob/nsites) %>%
  mutate(het_prop = (nsites-homo_ob)/nsites)
 
#plot inbreeding coefficient per individual
ggplot(ind_het, aes(f, fill = sex)) + geom_histogram(color = "black") + 
  theme_light() +
  ggtitle("Inbreeding coefficient per individual")

# plot heterozygosity proportion, in each sex
ggplot(ind_het) +
    geom_boxplot(aes(x = sex, y = het_prop, fill = sex)) + scale_y_continuous() + 
    theme_classic() +
    ggtitle("Proportion of heterzygotic sites in bees") +
      scale_y_continuous(expand=c(0,0), limits = c(0, 1))


# is there a significant difference in the proportion of heterozygotic sites between males and females?5
test <- ind_het %>%
  filter(sex == c("female", "male"))

wilcox.test(het_prop ~ sex, alternative = "two.sided", data = test)
t.test(asin(sqrt(het_prop)) ~ sex, alternative = "two.sided", data = test)

# no significant different (both wilcoxone and welch-test)
```

plot heterozygosity 
```{r}
# plot heterozygosity proportion, in each individual, color code by sex
ggplot(ind_het, aes(x=ind, y=het_prop, color=sex)) + 
    geom_point(size=3) +
    theme_classic() +
    ggtitle("Proportion of heterzygotic sites per bee") +
    xlab("Sample") + 
    ylab("Proportion of het zites")
```

### Heterozygosity per site
--hardy

Reports a p-value for each site from a Hardy-Weinberg Equilibrium test (as defined by Wigginton, Cutler and Abecasis (2005)). The resulting file (with suffix ".hwe") also contains the Observed numbers of Homozygotes and Heterozygotes and the corresponding Expected numbers under HWE.
#### for all individuals
```{r}
site_hwe <- read_delim("/Users/nuriteliash/Documents/GitHub/Variant_Calling_bees/data/vcf_stats/bee.hwe", delim = "\t", col_names = c("CHR","POS", "OBS(HOM1/HET/HOM2)", "E(HOM1/HET/HOM2)", "ChiSq_HWE", "P_HWE", "P_HET_DEFICIT", "P_HET_EXCESS"), skip = 1)

site_het <- site_hwe %>%
  select(CHR, POS, "OBS(HOM1/HET/HOM2)") %>%
  separate(col = "OBS(HOM1/HET/HOM2)", into = c("hom1", "het", "hom2"), sep = "\\/") %>%
  mutate_at(c("hom1", "het", "hom2"), as.numeric) %>%
  mutate(prop_het = (het/(hom1+ het+ hom2)))

site_het <-  mutate(site_het, site = paste(site_het$CHR, site_het$POS)) 
  
ggplot(site_het, aes(prop_het)) + geom_histogram(fill="#69b3a2", color="#e9ecef", alpha=0.8) + 
  theme_light() +
  ggtitle("Heterozygosity per site- all individuals")+
  xlab("Proportion of het genotype") + 
  ylab("Number of sites")
```

### Plot all samples Allele balance
```{r}
info <- read.csv("/Users/nuriteliash/Documents/GitHub/Variant_Calling_bees/data/meta.csv")
colnames(info) <-c("sample","sex")

vcf <- read.vcfR("/Users/nuriteliash/Documents/GitHub/Variant_Calling_bees/data/freebayes_filtered.recode.vcf", verbose = FALSE )
vcf

knitr::kable(vcf@gt[c(1:2,11,30),1:3])

# The ‘AD’ field in our VCF data includes the depth at which each allele was sequenced. We can extract this information with the function extract.gt().
ad <- extract.gt(vcf, element = 'AD')
knitr::kable(ad[c(1:2,11,30),1:6])

#The function extract.gt() isolates the ‘AD’ data from the colon delimited string in the VCF data. We expect integer counts for the number of sequences observed. However, because this data is comma delimited we need another step before we have integers. We use the function masplit() to extract the first and second allele. At that point we have integers so we can use math to create allele frequencies from the counts.
allele1 <- masplit(ad, record = 1)
allele2 <- masplit(ad, record = 2)

ad1 <- allele1 / (allele1 + allele2)
ad2 <- allele2 / (allele1 + allele2)

ad1M <- reshape::melt(ad1)
colnames(ad1M) = c("site", "sample", "value")
#add the sample names and info
#ad1M = ad1M %>% left_join(info, by = "sample") %>% replace(is.na(.),0)
# add the information that this is the reference allele
ad1M = ad1M %>%
  mutate(allele = "ref")

ad2M <- reshape::melt(ad2)
colnames(ad2M) = c("site", "sample", "value")
#add the sample names and info
#ad1M = ad1M %>% left_join(info, by = "sample") %>% replace(is.na(.),0)
ad2M = ad2M %>%
  mutate(allele = "alt")

table = bind_rows(ad1M, ad2M) %>%
   replace(is.na(.),0) %>% 
  left_join(info, by = "sample")
  
  
#plot adult males
 P_male_all= table %>%  dplyr::filter(sex =="male") %>%
 # filter(value>0) %>% # remove the very much homozygotic sites
  #filter(value<1) %>%
  ggplot( aes(x=value, fill= allele)) +
    geom_histogram(color="black", position = 'identity') +
        theme_classic() +
        theme(
      legend.position="none",
      panel.spacing = unit(0.1, "lines"),
      strip.text.x = element_text(size = 8)) +
    xlab("Allele balance") +
    ylab("Site count")+
  ggtitle("Bee Males Allele balance, all sites") +
  theme(legend.position="bottom")  +
    facet_wrap(~sample, scales = "free") 

P_male_no0_1 = table %>%  dplyr::filter(sex =="male") %>%
 filter(value>0) %>% # remove the very much homozygotic sites
  filter(value<1) %>%
  ggplot( aes(x=value, fill= allele)) +
    geom_histogram(color="black", position = 'identity') +
        theme_classic() +
        theme(
      legend.position="none",
      panel.spacing = unit(0.1, "lines"),
      strip.text.x = element_text(size = 8)) +
    xlab("Allele balance") +
    ylab("Site count")+
  ggtitle("Bee Males Allele balance, no '0' and '1'") +
  theme(legend.position="bottom")  +
    facet_wrap(~sample, scales = "free") 

#plot adult females
P_female_all= table %>%  dplyr::filter(sex =="female") %>%
 # filter(value>0) %>% # remove the very much homozygotic sites
  #filter(value<1) %>%
  ggplot( aes(x=value, fill= allele)) +
    geom_histogram(color="black", position = 'identity') +
        theme_classic() +
        theme(
      legend.position="none",
      panel.spacing = unit(0.1, "lines"),
      strip.text.x = element_text(size = 8)) +
    xlab("Allele balance") +
    ylab("Site count")+
  ggtitle("Bee Female Allele balance, all sites") +
  theme(legend.position="bottom")  +
    facet_wrap(~sample, scales = "free") 

P_female_no0_1 = table %>%  dplyr::filter(sex =="female") %>%
 filter(value>0) %>% # remove the very much homozygotic sites
  filter(value<1) %>%
  ggplot( aes(x=value, fill= allele)) +
    geom_histogram(color="black", position = 'identity') +
        theme_classic() +
        theme(
      legend.position="none",
      panel.spacing = unit(0.1, "lines"),
      strip.text.x = element_text(size = 8)) +
    xlab("Allele balance") +
    ylab("Site count")+
  ggtitle("Bee Female Allele balance, no '0' and '1'") +
  theme(legend.position="bottom")  +
    facet_wrap(~sample, scales = "free") 
```
